crates/ruff_python_parser/src/parser/expression.rs

```diff
@@ -459,37 +460,34 @@ impl<'src> Parser<'src> {
         let range = self.current_token_range();
 
         if self.at(TokenKind::Name) {
-            let (Tok::Name { name }, _) = self.bump(TokenKind::Name) else {
+            let TokenValue::Name(name) = self.take_value(TokenKind::Name) else {
```

Comment:
This is nice.

Reply:
By `TokenValueKind`, do you mean that instead of `TokenValue::None`, we'd have a `TokenValue::Other(TokenValueKind)`?

Reply:
What I had in mind is to add a `TokenValueKind` that has a variant for each variant in `TokenValue`. It would ensure that `take_value` is never called for a `TokenKind` that has no associated value. However, I'm no longer sure if it's worth the effort, considering that `take_value` also calls `bump`

Reply:
Yeah, I'm not sure either as it'll still involve converting to `TokenKind` because of the `bump` call.

Reply:
Yeah. The only benefit I see is that it would help to find e.g. all usages of `TokenValueKind::String`.